deku-c/wasm-vm-ocaml/env.ml

```diff
@@ -0,0 +1,229 @@
+                | Read_ticket handle ->
+                    let ticket_id, amount, handle =
+                      Ticket_table.read_ticket t.ticket_table handle
+                    in
+                    Some (fun k -> continue k (ticket_id, amount, handle))
+                | Mint_ticket (ticket_id, amount) ->
+                    let handle =
+                      Ticket_table.mint_ticket t.ticket_table ticket_id amount
+                    in
+                    Some (fun k -> continue k handle)
+                | Join_ticket (handle1, handle2) ->
+                    let handle : int option =
+                      Ticket_table.join_tickets t.ticket_table (handle1, handle2)
+                    in
+                    Some (fun k -> continue k handle)
+                | Split_ticket (handle, amounts) ->
+                    let handle =
+                      Ticket_table.split_ticket t.ticket_table handle amounts
+                    in
+                    Some (fun k -> continue k handle)
+                | Set_constants constants' ->
+                    constants := constants';
+                    Some (fun k -> continue k ())
+                | Set_instance instance ->
+                    (call_indirect :=
+                       fun idx arg ->
+                         let func =
+                           Externs.(extract_func instance Externs.call_callback)
+                           |> Result.get_ok
+                         in
+                         match
+                           Wasm.Eval.invoke func
+                             (Wasm.Values.(Num (I32 idx)) :: arg)
+                         with
+                         | Wasm.Values.[ Num (I64 x) ] -> x
+                         | _ -> failwith "bad indirect call in contract");
+                    (call_indirect_unit :=
+                       fun idx arg ->
+                         let func =
+                           Externs.(
+                             extract_func instance Externs.call_callback_unit)
+                           |> Result.get_ok
+                         in
+                         let _ =
+                           Wasm.Eval.invoke func
+                             (Wasm.Values.(Num (I32 idx)) :: arg)
+                         in
+                         ());
+                    (push :=
+                       fun arg ->
+                         let func =
+                           Externs.(extract_func instance Externs.push)
+                           |> Result.get_ok
+                         in
+                         ignore @@ Wasm.Eval.invoke func [ arg ]);
+                    Some (fun (k : (a, _) continuation) -> continue k ())
+                | _ -> None);
+          }
+    with
+    | Failure s -> Error ("unexpected failure. Failure " ^ s)
+    | Wasm.Eval.Crash _ | Wasm.Eval.Exhaustion _ | Wasm.Eval.Link _
+    | Wasm.Eval.Trap _ ->
+        Error "Invalid contract supplied to the vm"
```

Comment:
Don't ignore the actual errors, that makes debugging harder. I think something similar to line 53 is enough for now.